modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/DataSetSampleStatusClayTableDataSetFilter.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.sample.web.internal;
+
+import com.liferay.frontend.taglib.clay.data.set.filter.BaseCheckBoxClayDataSetFilter;
+import com.liferay.frontend.taglib.clay.data.set.filter.CheckBoxClayDataSetFilterItem;
+import com.liferay.frontend.taglib.clay.data.set.filter.ClayDataSetFilter;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.List;
+import java.util.Locale;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Javier de Arcos
+ */
+@Component(
+	property = "clay.data.set.display.name=dataSetDisplay",
+	service = ClayDataSetFilter.class
+)
+public class DataSetSampleStatusClayTableDataSetFilter
```

Comment:
How are these `*Filter`s and `*View`s wired to dataset display tag? Is it over `id` / `clay.data.set.display.name`?

I would expect them to be set either in `DisplayContext` or in a tag attribute. This is a bit too magicky for my taste :slightly_smiling_face:

Reply:
They are wired by the clay.data.set.display.name.

I agree with you that is a little obscure but I think it is a common pattern in some Liferay frameworks